packages/venia-ui/lib/components/Header/cartTriggerFragments.gql.js

```diff
@@ -0,0 +1,8 @@
+import gql from 'graphql-tag';
+
+export const CartTriggerFragment = gql`
+    fragment CartTriggerFragment on Cart {
+        id
+        total_quantity
```

Comment:
This ended up being interesting; `total_quantity` was already in `CartPageFragment`, so when I made this change, cart counter started updating w/o me including it explicitly in `CartPageFragment`. Is it okay to rely on that, or should we explicitly add `CartTriggerFragment` to every mutation as well?

Reply:
Is it possible that we were wrong about needing to re-use named mutations? Maybe it's hitting the cart cache because of the query signature. If that were the case fragment re-use would only be helpful in composed components where the parent requests all the children fragments. Cart Page, for example.

Reply:
Good question for sure.

We should ensure that mutations select any fields that may change as a result of that mutation in their response query; even if their coupled component does not need those fields for display.

If we do that I think it's perfectly fine to rely on the `CartPageFragment`'s selection of `total_quantity`. The cart page is saying both: "I need this field for my display" and "some of my mutations update this field".

If some day in the future the cart page no longer needs `total_quantity` for its display, it should still include it if any of its mutations result in `total_quantity` changing (specifically, the mutation whose action results in `total_quantity` changing should be sure to include it).

If no such mutation exists and `total_quantity` is removed entirely from `CartPageFragment`, we can be confident that `cartTrigger` will always be up to date because we know that nothing else is updating that underlying data, as @sirugh pointed out above.

Long story short, we should not add `CartTriggerFragment` to every mutation.

Reply:
> We should ensure that mutations select any fields that may change as a result of that mutation in their response query; even if their coupled component does not need those fields for display.

We're essentially establishing a subscription model here. We can accomplish that with refetch queries or by including data in the mutation for things that "may change". I'm not sure I agree with the inclusion of response data in a mutation. It's a little too magical. That said, the alternative is using refetchQueries which is essentially the same thing. 

"When something modifies this data, I need to know so I can update!"

In this case, we've got several components saying that they need to know about updates to some value. Which one is the right one? What if developers roll their own cart page instead of using ours?